sources/Interop/D3D12MemoryAllocator/include/D3D12MemAlloc.h.cs

```diff
@@ -0,0 +1,93 @@
+        /// </code>
+        /// </summary>
+        public static readonly uint D3D12MA_DEBUG_LEVEL = get_app_context_data(nameof(D3D12MA_DEBUG_LEVEL), IsDebug ? 2 : 0);
+
+        /// <summary>
+        /// When defined to value other than 0, the library will try to use
+        /// D3D12_SMALL_RESOURCE_PLACEMENT_ALIGNMENT or D3D12_SMALL_MSAA_RESOURCE_PLACEMENT_ALIGNMENT
+        /// for created textures when possible, which can save memory because some small textures
+        /// may get their alignment 4K and their size a multiply of 4K instead of 64K.
+        /// <code>
+        /// #define D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT 0
+        ///     Disables small texture alignment.
+        /// #define D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT 1
+        ///     Enables conservative algorithm that will use small alignment only for some textures
+        ///     that are surely known to support it.
+        /// #define D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT 2
+        ///     Enables query for small alignment to D3D12 (based on Microsoft sample) which will
+        ///     enable small alignment for more textures, but will also generate D3D Debug Layer
+        ///     error #721 on call to ID3D12Device::GetResourceAllocationInfo, which you should just
+        ///     ignore.
+        /// </code>
+        /// </summary>
+        public static readonly uint D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT = get_app_context_data(nameof(D3D12MA_USE_SMALL_RESOURCE_PLACEMENT_ALIGNMENT), 1);
+
+        /// <summary>
+        /// Minimum alignment of all allocations, in bytes.
+        /// Set to more than 1 for debugging purposes only.Must be power of two.
+        /// </summary>
+        public static readonly uint D3D12MA_DEBUG_ALIGNMENT = get_app_context_data(nameof(D3D12MA_DEBUG_ALIGNMENT), 1);
+
+        /// <summary>
+        /// Minimum margin before and after every allocation, in bytes.
+        /// Set nonzero for debugging purposes only.
+        /// </summary>
+        public static readonly uint D3D12MA_DEBUG_MARGIN = get_app_context_data(nameof(D3D12MA_DEBUG_MARGIN), 0);
+
+        /// <summary>
+        /// Set this to 1 for debugging purposes only, to enable single mutex protecting all
+        /// entry calls to the library.Can be useful for debugging multithreading issues.
+        /// </summary>
+        public static readonly uint D3D12MA_DEBUG_GLOBAL_MUTEX = get_app_context_data(nameof(D3D12MA_DEBUG_GLOBAL_MUTEX), 0);
+
+        /// <summary>Define this macro to 0 to disable usage of DXGI 1.4 (needed for IDXGIAdapter3 and query for memory budget).</summary>
+        public static readonly uint D3D12MA_DXGI_1_4 = get_app_context_data(nameof(D3D12MA_DXGI_1_4), 1);
+
+        /// <summary>Number of D3D12 memory heap types supported.</summary>
+        [NativeTypeName("UINT")] internal const uint HEAP_TYPE_COUNT = 3;
```

Comment:
nit: attributes on a separate line, except when parameters are involved

```suggestion
        [NativeTypeName("UINT")]
        internal const uint HEAP_TYPE_COUNT = 3;
```

Reply:
This also looks too be in the header file and is therefore normally public.

It would likewise be good to make it `static readonly` so it can be changed without breaking users in the future if they depend on it.

Reply:
Fixed in 04e5f82de4c12a7af1ca4da7673c126e11e14baa 🙂
Question - should I do this for fields too? And if so, do I then insert empty spaces between them?
For now they're just inlined, eg.:

https://github.com/terrafx/terrafx.interop.d3d12memoryallocator/blob/04e5f82de4c12a7af1ca4da7673c126e11e14baa/sources/Interop/D3D12MemoryAllocator/src/Allocation.cs#L24-L30

Reply:
Yes please.

Reply:
Done in 24863744c722f296bec598e5df6ca3368472b227, hope this is alright 🙂